### PR TITLE
Feature/integration testing nw

### LINKED ISSUE #28: Add propTypes
Need to add propTypes to all files

### LINKED ISSUE #37: Testing: Reducers
Create a test for all reducers: 

- [ ] books

- [ ] haveRead

- [ ] list

- [ ] toRead

- [ ] userGoal

### PR DESCRIPTION
**Pull request by nware1066**
### Description

Added Proptypes for all components and Unit testing for reducers

### Types of changes:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features

### Has This Been Tested?
- [ ] No
- [x] Yes

### What are the relevant tickets?

closes #37 closes #28 #14 #33 

### Message/Questions for reviewer:

Test for GoalForm in App is not complete or passing, this will need to be completed

### Checklist:
- [ ] My code follows the code style of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/reducers/reducers.test.js [hunk 1]
```diff
@@ -73,3 +73,48 @@ describe('userGoal', () => {
     expect(result).toEqual(expected);
   })
 })
+
```
Comment (aemcdonald) [inline src/reducers/reducers.test.js:76]: Great job wrapping up our reducer tests here

**Review by aemcdonald — APPROVED** [review]
I agree with you regarding moving the test for setting a user goal into App. It is more fitting there as an integration test so that we can test that the goal is then displayed on the page after a user clicks submit

File: src/Components/ToRead/ToRead.js [hunk 2]
```diff
@@ -36,3 +37,7 @@ const mapStateToProps = state => ({
 export default connect(
   mapStateToProps
 )(ToRead);
+
+ToRead.propTypes = {
+  toReadList: PropTypes.array
+}
```
Comment (stacyp2006) [inline src/Components/ToRead/ToRead.js:43]: Great job getting these PropTypes added

**Review by stacyp2006 — APPROVED** [review]
Looks good. We're getting close to the finish line

**Pull request merged**